packages/excalidraw/tests/fractionalIndex.test.ts

```diff
@@ -0,0 +1,329 @@
+      movedOneLeftFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    );
+
+    testValidity(movedOneRightFixedElements);
+
+    randomlySelected = [
+      ...new Set([
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+      ]),
+    ];
+
+    const movedAllLeftFixedElements = moveAllLeft(
+      movedOneRightFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    ) as ExcalidrawElement[];
+
+    testValidity(movedAllLeftFixedElements);
+
+    randomlySelected = [
+      ...new Set([
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+      ]),
+    ];
+
+    const movedAllRightFixedElements = moveAllLeft(
+      movedAllLeftFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    ) as ExcalidrawElement[];
+
+    testValidity(movedAllRightFixedElements);
+  });
+});
```

Comment:
Could you please add two more test cases, creating max number of elements let's say 10-20k (one by one), while on each addition creating a fractional index. Adding element at the end is our default behaviour, which essentially leads to left-skewed indices due to halving. We should not have problems with length as the upper bound (as opposed to IEEE754 variation), but still I am interested how long the indices could get:
- length of last non-jittered index - for dev I would even prefer more readable base36 character set
- length of last jittered index ending with default base62 set (should be much shorter, but what jitter size do we expect - i.e. could we afford to extend it to limit collisions?)

Reply:
- for adding to the end, there's not halving involved, since its `generateKeyBetween(key, null)` and this will just "increment" based on the given key 
- using base36 for dev now
- changed jitter range from 3 to 6

Reply:
> for adding to the end, there's not halving involved, since its generateKeyBetween(key, null) and this will just "increment" based on the given key

Interesting!

Reply:
You also specified the character set for prod to be base62, does it mean we now also have base62 jitter? Meaning 62^6 ~ one in a 56 billion chance for collision? Asking since based on the description their char set for jitter is base36 -> 36^3 ~ 47K.

![image](https://github.com/excalidraw/excalidraw/assets/20387666/0eeaa0ba-41b4-4609-b00b-6295230308a7)

Reply:
> You also specified the character set for prod to be base62, does it mean we now also have base62 jitter? Meaning 62^6 ~ one in a 56 billion chance for collision? Asking since based on the description their char set for jitter is base36 -> 36^3 ~ 47K (which seems odd).
> 
> ![image](https://private-user-images.githubusercontent.com/20387666/291683171-0eeaa0ba-41b4-4609-b00b-6295230308a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjAxMzYsIm5iZiI6MTcwMzA1OTgzNiwicGF0aCI6Ii8yMDM4NzY2Ni8yOTE2ODMxNzEtMGVlYWEwYmEtNDFiNC00NjA5LWIwMGItNjI5NTIzMDMwOGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4MTAzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3YWIwZThmZmMyZGRjZDcxZTBkYWZmN2RlZTgyYjVlNzBkYmRlNmNhYWNkM2Q4YmQ5M2FiM2MyY2JmNGVhNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.c80kZEKwAH4dJa5XDVO2B8e3jK1q8pLzDxfxtioazxM)

From the source code, they are definitely using base62. So I'm not sure how they arrived at `47.000`. We could also reduce the jittering length if 6 jittering length is too much storage space/bandwidth for our scale.

Reply:
Let's put 4, If it's base62 jitter, it should be more than enough.